Lib/_osx_support.py

```diff
@@ -331,6 +345,13 @@ def compiler_fixup(compiler_so, cc_args):
             except ValueError:
                 break
 
+    elif not _supports_arm64_builds():
+        # Look for "-arch arm64" and drop that
+        for idx in range(len(compiler_so)):
+            if compiler_so[idx] == '-arch' and compiler_so[idx+1] == "arm64":
+                del compiler_so[idx:idx+2]
+                break
```

Comment:
Can there be multiple occurrences of `-arch arm64`?

Reply:
Only when the user has specified the same architecture multiple times.  The compiler ignores multiple occurrences of the same architecture.  I'll drop the "break".

Reply:
I don't think that the user will do this intentionally, but I afraid that in generated command lines it can happen. It happens with some commands in our Makefile.

Dropping `break` is not correct because `del` shifts the content. You can write the code in the similar ways to the loop above (lines 340-346). Or iterate in the opposite direction: ` for idx in range(len(compiler_so))[::-1]`, it may be simpler.